### ISSUE TITLE
Add `Execution` to the `types.ExecuteRequest` struct

### ISSUE THREAD

**Issue by cobolbaby** [dkron-io/dkron#1307]
**Is your feature request related to a problem? Please describe.**

I wish to add `Execution` metadata to the `reportingWriter` struct to implement some extended features. However, to achieve this, it is necessary to adjust the structure of the ExecuteRequest. I am not sure whether such a request is acceptable.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

https://github.com/distribworks/dkron/blob/53864a0cf5567e31305b5b004474fb546062eb09/dkron/grpc_agent.go#L84-L94

**Comment by vcastellm** [comment]
Can you explain what metadata do you want to include?

**Comment by cobolbaby** [comment]
> Can you explain what metadata do you want to include?

Currently I want to get the following three attributes.

```
type Execution struct {
	// Id is the Key for this execution
	Id string `json:"id,omitempty"`
        ....
	// Start time of the execution.
	StartedAt time.Time `json:"started_at,omitempty"`
        ....
	// Node name of the node that run this execution.
	NodeName string `json:"node_name,omitempty"`
       ....
}
```

**Comment by vcastellm** [comment]
@cobolbaby sorry for the delay, can you let us know if you're still interested in this?

**Comment by cobolbaby** [comment]
Yes, this is an important improvement that facilitates integration with other systems.
